### ISSUE TITLE
POP OS Bluetooth Headset options not available (HSP/HFP - mSBC)

### ISSUE THREAD

**Issue by SuperNolvas** [pop-os/pop#2740]
### Discussed in https://github.com/pop-os/pop/discussions/2739

<div type='discussions-op-text'>

<sup>Originally posted by **SuperNolvas** December 30, 2022</sup>
Hello. I have noticed that POP OS defaults to AD2P ONLY for any Bluetooth headset connection meaning any microphone input is not available. The option for Handsfree is not available at all and can't be enabled. The handsfree options usually give HSP/HFP , codec mSBC or HSP/HFP, codec CVSD. I have tried editing Wireplumber config and Pipewire but to no avail. I also tested Fedora on the same hardware and this worked on the Live ISO but when installed reverted to the behaviour POP OS is showing. I also tried Garuda Linux (Arch based) on this hardware and it behaved corrected and gave Headset/Handsfree options and also configuration options in the Output section for CODECS for AD2P. I also tested Ubuntu 22.10 on my laptop so different hardware to the PC and it works fine. I do know that my ear buds work correctly and my bluetooth hardware works correctly as I have proven this by using Garuda and Ubuntu. 

My PC is Ryzen 5 3600
GTX 3060 12GB
Intel Wireless AC 9260 with Bluetooth 5.1

To show a comparison I will first show screenshots of POP OS illustrating the limited option of AD2P. Then I will show screenshots of Ubuntu with the expected behaviour. I have also made a pipewire configuration dump using the command line program pw-dump on both operating systems for comparison if that helps with the troubleshooting. I have also included the output of patcl list cards for each OS. 

**POP OS Headset ONLY, Handsfree not available**

![POS_OS_Headset_only](https://user-images.githubusercontent.com/17798480/210107830-cf70acda-b4be-4230-9d5d-e336c5d8ce01.png)

![POP_OS_Output_setting](https://user-images.githubusercontent.com/17798480/210107910-e20bcb8e-5238-4897-865d-ea3807107e13.png)

**POP OS - pactl list cards output**

![POP_OS_patcl](https://user-images.githubusercontent.com/17798480/210107918-511ff39f-2664-423f-b0f5-70ba8592a271.png)

**POP OS Pipewire config dump**

[pop_os_22.04_pipewire_config.txt](https://github.com/pop-os/pop/files/10326135/pop_os_22.04_pipewire_config.txt)

**Ubuntu Headset and Handsfree Input / Output settings - Expected behaviour which is missing from POP OS**

![Ubuntu_Headset](https://user-images.githubusercontent.com/17798480/210108121-d05caf7a-649e-4e54-b37d-626b84f627b1.png)
tu 22.10 

![Ubuntu_Headset_Handsfree](https://user-images.githubusercontent.com/17798480/210108154-f52585b4-2309-43ad-8a4b-203c7f64c3de.png)

![Ubuntu_Output_configuration](https://user-images.githubusercontent.com/17798480/210108162-c0a69f66-3509-4f65-a7a7-18c4d5c38f41.png)

**Ubuntu pactl list cards output**

![Ubuntu_pactl](https://user-images.githubusercontent.com/17798480/210108176-b86e5dbb-99b0-490f-8849-bc04c7dfd51c.png)

**Ubuntu pipewire config**

[ubuntu_22.10_pipewire_config.txt](https://github.com/pop-os/pop/files/10326145/ubuntu_22.10_pipewire_config.txt)

Lastly I have looked at pipwire version, bluez version and wireplumber versions and they differ between Ubuntu, POP OS and Garuda. Garuda having all the latest versions, POP OS some newer/older and Ubuntu some newer/older. So I haven't found any common rationale for this not working but I have seen the same behaviour in other operating systems. I believe this is arising because of a combination of bluez, pipewire and wireplumber configurations but the answer to what is wrong is beyond my troubleshooting capabilities. I would be grateful if the community could look at this issue and I can provide further logs from POP OS and Ubuntu if required. 

</div>

**Comment by SuperNolvas** [comment]
Fixed as of 26/01/2023 updates to Pipewire.

**Comment by crostine** [comment]
i have the inverse problem. i want my headphone on ad2p but pop os no longer lists it as an option. i dont know what changed or why but there was never a problem with this before, since a couple days my laptop simply cant do ad2p anymore apparently.. the only option left over is "Handsfree - <headphone name>" but this is a mono output with a lot of distortion.
the headphone is fine though it works perfectly with other devices like my phone or windows laptop.

**Comment by zeldoon** [comment]
So I tried all kinds of crazy shit to get this to work... short of switching bluez and got something to work. Please try this and let me know if it helps you.

in /etc/bluetooth/main.conf

on line 5, uncomment: Name = BlueZ

**Comment by regtm** [comment]
> Fixed as of 26/01/2023 updates to Pipewire.

Any indication what fixed it? I seem to have the same issue.

**Comment by claytronOnLinux** [comment]
From a thread found on reddit whic advized I run

sudo apt reinstall pipewire pipewire-pulse
systemctl --user --now enable pipewire pipewire-pulse
reboot

Combined with zeldoon's comment

> So I tried all kinds of crazy shit to get this to work... short of switching bluez and got something to work. Please try this and let me know if it helps you.
> 
> in /etc/bluetooth/main.conf
> 
> on line 5, uncomment: Name = BlueZ

Then log out and log in

I was able to progress through my problems of not being able to connect, then being able to connect but the sound was very low quality, to being able to connect with high quality audio again.

**Comment by reubenbrown13** [comment]
I have tried the suggestions, but my Bluetooth headset still shows no microphone, and bluman shows only a2dp profiles.  Any other ideas?

**Comment by aacanakin** [comment]
> > Fixed as of 26/01/2023 updates to Pipewire.
> 
> Any indication what fixed it? I seem to have the same issue.

Same for me, the problem still exists, headset option is not available specifically when headphones are connected automatically

**Comment by SuperNolvas** [comment]
> > > Fixed as of 26/01/2023 updates to Pipewire.
> > 
> > 
> > Any indication what fixed it? I seem to have the same issue.
> 
> Same for me, the problem still exists, headset option is not available specifically when headphones are connected automatically

I don't know sorry, by March 2023 I had moved house and I didn't need to use BT headphones anymore. Maybe try deleting the bluetooth folder as reubenbrown13 suggests above.

**Comment by keithfancher** [comment]
> Same for me, the problem still exists, headset option is not available specifically when headphones are connected automatically

Same here. Seems to be the inverse of @SuperNolvas original issue. For me, *only* handsfree is available, headset is *not* available. Which of course means very low quality mono sound.

Interestingly, for me this only started relatively recently. Maybe a few weeks ago? Can't pinpoint the exact date/update.

I can delete the pairing and re-pair and it will work as expected **once**, but the next time I connect there is no longer a "headset" option. I have to delete and re-pair *every time* I use my headphones.

PopOS 22.04. All latest updates installed. I've tested the same headphones on Ubuntu, no issue there.

**Comment by reubenbrown13** [comment]
If it helps, here are the contents for my bash script that I use to make this easier for the time being.  I still have to re-pair the headset to the computer every time I want to use the handsfree/mic, but the headphones work just fine w/o having to do this once paired.  Obviously you will need to change the folder name for your folder.

What I did was copy the cached folder when the handsfree was working, so I could then restore it to the Bluetooth folder and then restart the service to get the config to load.

`sudo rm -R /var/lib/bluetooth/BC\:03\:58\:F5\:8C\:2C/
sudo cp -R  ~/BC\:03\:58\:F5\:8C\:2C/ /var/lib/bluetooth/BC\:03\:58\:F5\:8C\:2C/
sudo systemctl restart bluetooth`

**Comment by keithfancher** [comment]
I think I tracked down the issue! See my comment on https://github.com/pop-os/pop/issues/3347#issuecomment-2339377213.

TL;DR: At least the issue that *some* of us are experiencing was likely introduced in a kernel update on 6/20/2024. Downgrading to kernel version `6.9.2` supposedly solves the problem, though I don't know how easy that is to do on PopOS.

**Comment by srosato** [comment]
I also started to experience this on my bluetooth Sennheiser momentum 3 true wireless headset. It started happening after I did a kernel update last May 30th. 

If I pair for the first time it works. If I close my headset and re-open it, it only connects to the headset unit on codec mSBC. If I restart bluetooth, sometimes it connects back to the High Fidelity Playback. If I close my headset again and re-open it, it will glitch again.

@keithfancher I followed your comment up to the issue you mentioned and provided additional log info on their GitLab in the hope it can help

**Comment by madscientist42** [comment]
Sony Noise Cancellers with a Mic mode, an ONN air pods clone in my case.  _**Should**_ be able to gear shift and doesn't.

The problem begins with selecting the handsfree.  The moment that is selected, the configuration menu items (Which were in a counterintuitive place...by the by...(That _**should**_ be a main item off of the bar there...  ;) ) lose their ever loving mind.  During the current connection it sort-of works, sort of doesn't.  When you reconnect, the session **_forever_** seems bound to the headset mode...which is broken.  Only ways to un-pooch the thing is to un/re-pair the device in question, or use Blueman to force it to switch modes properly.  Seems there is a solid disconnect with what you're trying to do and what needs to be done with Bluez/Pipewire there- that Blueman understands properly.

More than happy to work with someone in Aurora (Or wherever they're based if not in Colorado) to help sort this out and at least get it fixed within the Alpha/Betas of the New product.  It's a nuisance that y'all have let go for two plus years now.  You want to be considered to be serious...this was one of those things that hamper that image  ;)
